pkg/resources/external_oauth_integration.go

```diff
@@ -41,548 +46,777 @@ var oauthExternalIntegrationSchema = map[string]*schema.Schema{
+	if prop, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool { return property.Name == "COMMENT" }); err == nil {
+		if err = d.Set("comment", prop.Value); err != nil {
+			return nil, err
+		}
 	}
-	// if err := d.Set("created_on", showOutput.CreatedOn.String); err != nil {
-	// 	return fmt.Errorf("error setting created_on: %w", err)
-	// }
+	return []*schema.ResourceData{d}, nil
+}
 
-	// DESCRIBE
-	stmt, err = manager.ReadDescribe(input)
+func CreateContextExternalOauthIntegration(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	enabled := d.Get("enabled").(bool)
+	externalOauthIssuer := d.Get("external_oauth_issuer").(string)
+	externalOauthSnowflakeUserMappingAttributeRaw := d.Get("external_oauth_snowflake_user_mapping_attribute").(string)
+	externalOauthTokenUserMappingClaimRaw := expandStringList(d.Get("external_oauth_token_user_mapping_claim").(*schema.Set).List())
+	name := d.Get("name").(string)
+	integrationTypeRaw := d.Get("external_oauth_type").(string)
+	integrationType, err := sdk.ToExternalOauthSecurityIntegrationTypeOption(integrationTypeRaw)
 	if err != nil {
-		return fmt.Errorf("couldn't generate describe statement: %w", err)
+		return diag.FromErr(err)
 	}
-
-	rows, err := snowflake.Query(db, stmt)
+	externalOauthSnowflakeUserMappingAttribute, err := sdk.ToExternalOauthSecurityIntegrationSnowflakeUserMappingAttributeOption(externalOauthSnowflakeUserMappingAttributeRaw)
 	if err != nil {
-		return fmt.Errorf("error querying external oauth integration: %w", err)
+		return diag.FromErr(err)
 	}
-
-	defer rows.Close()
-	describeOutput, err := manager.ParseDescribe(rows.Rows)
-	if err != nil {
-		return fmt.Errorf("failed to parse result of describe: %w", err)
+	externalOauthTokenUserMappingClaim := make([]sdk.TokenUserMappingClaim, 0, len(externalOauthTokenUserMappingClaimRaw))
+	for _, v := range externalOauthTokenUserMappingClaimRaw {
+		externalOauthTokenUserMappingClaim = append(externalOauthTokenUserMappingClaim, sdk.TokenUserMappingClaim{Claim: v})
 	}
+	id := sdk.NewAccountObjectIdentifier(name)
+	req := sdk.NewCreateExternalOauthSecurityIntegrationRequest(id, enabled, integrationType, externalOauthIssuer, externalOauthTokenUserMappingClaim, externalOauthSnowflakeUserMappingAttribute)
 
-	if err := d.Set("issuer", describeOutput.ExternalOauthIssuer); err != nil {
-		return fmt.Errorf("error setting issuer: %w", err)
-	}
-	if err := d.Set("jws_keys_urls", describeOutput.ExternalOauthJwsKeysURL); err != nil {
-		return fmt.Errorf("error setting jws_keys_urls: %w", err)
+	if v, ok := d.GetOk("comment"); ok {
+		req.WithComment(v.(string))
 	}
-	if err := d.Set("any_role_mode", describeOutput.ExternalOauthAnyRoleMode); err != nil {
-		return fmt.Errorf("error setting any_role_mode: %w", err)
+
+	if v, ok := d.GetOk("external_oauth_allowed_roles_list"); ok {
```

Comment:
I don't remember, but aren't sets always ok == true? If yes, then we have to be careful if CREATE supports empty allowed_roles_list, but I assume in this case it's supported (If I'm right).

Reply:
It returns false in this case.